app/uk/gov/hmrc/personaldetailsvalidation/endpoints/PersonalDetailsCollectionController.scala

```diff
@@ -176,6 +186,22 @@ class PersonalDetailsCollectionController @Inject()(personalDetailsSubmission: P
     }
   }
 
+  private def sessionKeyCheck(result: Future[Result], completionUrl: CompletionUrl, failureUrl: Option[CompletionUrl])(implicit request: MessagesRequest[AnyContent]): Future[Result] = {
+    val missingSessionKeys = List(FIRST_NAME_KEY, LAST_NAME_KEY, DOB_KEY).map(
```

Comment:
getting the sessions keys already has a method "retrieveMainDetails" see L176, do we need to reinvent the wheel? or can we follow the pattern of 176 to make it clearer that those are the "mainDetails"
Perhaps you could also rename the "retrieveMainDetails" to be something even more clear. "sessionRetrievalFirstPage"